src/youtube/js/youtube.js

```diff
@@ -0,0 +1,394 @@
+  var bitrateFlag = true;
+  var javascriptCommandQueue = [];
+  var qualities = [];
+
+  /**
+   * @class OoyalaYoutubeVideoFactory
+   * @classdesc Factory for creating video player objects for youtube videos.
+   * @property {string} name The name of the plugin
+   * @property {object} encodings An array of supported encoding types (ex. OO.VIDEO.ENCODING.MP4)
+   * @property {object} features An array of supported features (ex. OO.VIDEO.FEATURE.BITRATE_CONTROL)
+   * @property {string} technology The core video technology (ex. OO.VIDEO.TECHNOLOGY.HTML5)
+   */
+  var OoyalaYoutubeVideoFactory = function() {
+    this.name = pluginName;
+    this.features = [ OO.VIDEO.FEATURE.VIDEO_OBJECT_SHARING_GIVE,
+                      OO.VIDEO.FEATURE.BITRATE_CONTROL ];
+    this.technology = OO.VIDEO.TECHNOLOGY.HTML5;
+    this.encodings = [OO.VIDEO.ENCODING.MP4];
+
+    /**
+     * Creates a video player instance using OoyalaYoutubeVideoWrapper.
+     * @public
+     * @method OoyalaYoutubeVideoFactory#create
+     * @param {object} parentContainer The jquery div that should act as the parent for the video element
+     * @param {string} domId The id of the video player instance to create
+     * @param {object} controller A reference to the video controller in the Ooyala player
+     * @param {object} css The css to apply to the video element
+     * @param {string} playerId An id that represents the player instance
+     * @returns {object} A reference to the wrapper for the newly created element
+     */
+    this.create = function(parentContainer, domId, controller, css, playerId) {
+      if(player == null || parentContainer == null || controller == null)
+      {
+        console.warn("Youtube: Failed to create the player");
+        return;
+      }
+      youtubeVideoContainer = parentContainer;
+      $('head').append("<script type = 'text/javascript' src = 'http://www.youtube.com/iframe_api'></script>");
+      window.onYouTubePlayerAPIReady = function() { onYouTubeIframeAPIReady(); }; 
+      element = new OoyalaYoutubeVideoWrapper();
+      if(element == null) return;
+      element.controller = controller;
+      controller.notify(controller.EVENTS.CAN_PLAY);
+
+      youtubeVideoContainer.append(player);
+      return element;
+    };
+
+    /**
+     * Destroys the video technology factory
+     * @public
+     * @method OoyalaVideoFactory#destroy
+     */
+    this.destroy = function() {
+      this.encodings = [];
+      this.create = function() {};
+    };
+
+    this.maxSupportedElements = -1;
+  };
+
+  /* 
+   *This function creates an <iframe> (YouTube player)
+   *after the API code downloads.
+   * @property {object} youtubePlayer The youtube player is creted and assigned to youtubePlayer.
+   */
+  function onYouTubeIframeAPIReady() {
+    youtubePlayer = new YT.Player('player', {
+      videoId: youtubeID,
+      height: "100%",
+      width: "100%",
+      playerVars: { 'autoplay': 0, 'controls': 0 },
```

Comment:
No. These are youtube parameters which should not be allowed to modify. As we are not using youtube controls, the value of controls should always be zero and the value of autoplay should also be zero as we are using the Ooyala V4 skin which has its own autoplay parameter.